<a href="https://colab.research.google.com/github/swarup7350/python_code/blob/master/Analysis%20Of%20HIMS(16-01-2020).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:

import sys
import pymongo                                                                      # Import packages
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.pyplot import pie, axis, show
%matplotlib inline                                                              

import warnings                                                                 # Ignore warning related to pandas_profiling
warnings.filterwarnings('ignore') 
pd.set_option('display.max_columns', 100)                                       # Display all dataframe columns in outputs

In [0]:

from google.colab import files
uploaded = files.upload()

Saving Screening Final Details.xlsx to Screening Final Details (2).xlsx


In [0]:
import pandas as pd
Screening_Final_Details= pd.read_excel('Screening Final Details (1).xlsx', sep=',', encoding='latin-1')

In [92]:
Screening_Final_Details.head()

CampId            CampNo  \
0     108  CAMP_482_0000019   
1     108  CAMP_482_0000019   
2     108  CAMP_482_0000019   
3     108  CAMP_482_0000019   
4     108  CAMP_482_0000019   

                                        CampLocation   CampDate  DISTLGDCODE  \
0  Dattapada Rd, Rajendra Nagar, Borivali East, M... 2019-08-01          999   
1  Dattapada Rd, Rajendra Nagar, Borivali East, M... 2019-08-01          999   
2  Dattapada Rd, Rajendra Nagar, Borivali East, M... 2019-08-01          999   
3  Dattapada Rd, Rajendra Nagar, Borivali East, M... 2019-08-01          999   
4  Dattapada Rd, Rajendra Nagar, Borivali East, M... 2019-08-01          999   

               DISTNAME  RegdId        RegdNo         FULLNAME        MOBILE  \
0  MUMBAI SUBURBAN WEST  476802  353690015132    Kathuao Malik  8.828341e+09   
1  MUMBAI SUBURBAN WEST  476806  353690009820   Sudhansu Malik  9.892363e+09   
2  MUMBAI SUBURBAN WEST  476809  353690015123  Jitrendra Malik  8.452993e+09   
3  MUMBAI SUBURBAN WEST  476811  353690013716  Madhu Sudan Das  9.768925e+09   
4  MUMBAI SUBURBAN WEST  476813  353690008804        Anup Bain  9.082932e+09   

    AGE GENDER                                  DOCREMARK health_STATUS  \
0  28.0      M  PATIENT IS NORMAL HAEMODYNAMICALLY STABLE        Normal   
1  23.0      M  PATIENT IS NORMAL HAEMODYNAMICALLY STABLE        Normal   
2  20.0      M  PATIENT IS NORMAL HAEMODYNAMICALLY STABLE        Normal   
3  51.0      M  PATIENT IS NORMAL HAEMODYNAMICALLY STABLE        Normal   
4  41.0      M                                PREDIABETIC           IPD   

       BMI  Weight_KGs  Height_CMs  BloodSugar_F  BloodSugar_PP  BloodSugar_R  \
0  18.5901        50.0       164.0         101.0          115.0           NaN   
1  18.8189        50.0       163.0          71.0           96.0           NaN   
2  17.5781        45.0       160.0          74.0          148.0           NaN   
3  19.2277        48.0       158.0          83.0          125.0           NaN   
4  22.8624        60.0       162.0         100.0          166.0           NaN   

   PulseRate  
0        NaN  
1        NaN  
2        NaN  
3        NaN  
4        NaN

In [0]:
Screening_Final_Details.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64023 entries, 0 to 64022
Data columns (total 21 columns):
CampId           64023 non-null int64
CampNo           64014 non-null object
CampLocation     64014 non-null object
CampDate         64023 non-null datetime64[ns]
DISTLGDCODE      64023 non-null int64
DISTNAME         64023 non-null object
RegdId           64023 non-null int64
RegdNo           64023 non-null int64
FULLNAME         64021 non-null object
MOBILE           53356 non-null float64
AGE              64021 non-null float64
GENDER           64021 non-null object
DOCREMARK        54950 non-null object
health_STATUS    64023 non-null object
BMI              63899 non-null float64
Weight_KGs       63899 non-null float64
Height_CMs       63899 non-null float64
BloodSugar_F     57872 non-null float64
BloodSugar_PP    62887 non-null float64
BloodSugar_R     6558 non-null float64
PulseRate        0 non-null float64
dtypes: datetime64[ns](1), float64(9), int64(4), object(7)
memor

In [95]:
Screening_Final_Details.describe()

CampId   DISTLGDCODE         RegdId        RegdNo        MOBILE  \
count  64023.000000  64023.000000   64023.000000  6.402300e+04  5.335600e+04   
mean     457.211690    507.417928  510527.469519  1.897175e+11  8.845446e+09   
std      261.691495    107.817574   39964.263560  1.166427e+11  9.842298e+08   
min       67.000000    466.000000  240163.000000  0.000000e+00  0.000000e+00   
25%      227.000000    478.000000  490359.500000  1.110500e+11  8.108219e+09   
50%      410.000000    484.000000  518567.000000  1.717800e+11  9.096865e+09   
75%      715.000000    490.000000  536030.500000  2.930401e+11  9.637762e+09   
max      915.000000    999.000000  553332.000000  1.110500e+12  1.000000e+10   

                AGE           BMI    Weight_KGs    Height_CMs   BloodSugar_F  \
count  64021.000000  6.389900e+04  63899.000000  63899.000000   57872.000000   
mean      37.224786  2.870230e+03     60.342666    153.635766     115.146452   
std       11.931277  1.505336e+05     92.381559     37.359442    1056.567110   
min        0.000000  0.000000e+00      0.000000      0.160000       0.000000   
25%       30.000000  1.948700e+01     50.000000    152.000000      90.000000   
50%       36.000000  2.220630e+01     56.000000    161.000000      99.000000   
75%       45.000000  2.540280e+01     65.000000    168.000000     111.000000   
max      930.000000  2.884153e+07   8565.000000    768.000000  130162.000000   

       BloodSugar_PP   BloodSugar_R  PulseRate  
count   62887.000000    6558.000000        0.0  
mean      130.780447     141.784811        NaN  
std        34.672930    1376.534018        NaN  
min         0.000000       0.000000        NaN  
25%       112.000000     102.000000        NaN  
50%       128.000000     117.000000        NaN  
75%       142.000000     138.000000        NaN  
max       968.000000  110120.000000        NaN

In [166]:
Screening_Final_Details.groupby('CampDate')['RegdId'].count()

CampDate
2019-06-22     123
2019-06-29     118
2019-07-04     133
2019-07-05      84
2019-07-06     217
              ... 
2020-01-11    1209
2020-01-12     462
2020-01-13    1273
2020-01-14    1136
2020-01-15       1
Name: RegdId, Length: 145, dtype: int64

Date Wise Count of Registration 

In [113]:
Screening_Final_Details.groupby('GENDER')['RegdId'].count()

GENDER
F    21059
M    42962
Name: RegdId, dtype: int64

1) Total 21059 Females and 42962 Male are Participets in this camping.


In [114]:
Screening_Final_Details.groupby('health_STATUS')['RegdId'].count()

health_STATUS
CRITICAL     9062
IPD         12602
Normal      42359
Name: RegdId, dtype: int64

1) From Male and Female 42359 having Normal Health coundition, 12602 having IPD and 9062 having Critical coundition.

In [144]:
pd.crosstab(Screening_Final_Details['DISTNAME'],Screening_Final_Details['GENDER']).head(5)

GENDER        F     M
DISTNAME             
AHMEDNAGAR   41   234
AKOLA        47   679
AMRAVATI    771  1550
AURANGABAD   25   136
BEED         39   271

DISTRICT wise MALE and FEMALE Count

In [145]:
pd.crosstab(Screening_Final_Details['health_STATUS'],Screening_Final_Details['GENDER'])

GENDER             F      M
health_STATUS              
CRITICAL        3855   5205
IPD             4393   8209
Normal         12811  29548

HELTH STATUS GENDER wise count

In [162]:
pd.crosstab(Screening_Final_Details['Weight_KGs'],Screening_Final_Details['GENDER']).head(10)

GENDER       F   M
Weight_KGs        
0.0          0   2
2.0          1   2
4.0         10   9
5.0          6  14
5.5          1   1
6.0          4   9
7.0          4   9
8.0          0   1
9.0          0   1
12.0         0   1

How It was possible someone having 0 KGs Weight 

In [161]:
pd.crosstab(Screening_Final_Details['Weight_KGs'],Screening_Final_Details['GENDER']).tail()

GENDER      F  M
Weight_KGs      
6522.0      0  1
7014.0      0  1
7057.0      0  1
7525.0      1  0
8565.0      0  1

Very High Values in Weight

In [0]:
df=pd.pivot_table(Screening_Final_Details,index=["health_STATUS"],values=["RegdId"],
               columns=["DISTNAME"],aggfunc=[len],fill_value=0,margins=True)

In [0]:
df.head()

1.

In [0]:
df=pd.pivot_table(Screening_Final_Details,index=["health_STATUS"],values=["RegdId"],
               columns=["GENDER"],aggfunc=[len],fill_value=0,margins=True)

In [85]:
df.head(11)

len              
              RegdId              
GENDER             F      M    All
health_STATUS                     
CRITICAL        3855   5205   9060
IPD             4393   8209  12602
Normal         12811  29548  42359
All            21059  42962  64021

SyntaxError: ignored

In [0]:
df.to_excel('result.xlsx')

In [81]:
!pip install pandas_profiling
import seaborn